code/modules/integrated_electronics/subtypes/converters.dm

```diff
@@ -168,6 +202,56 @@
 
 	activate_pin(2)
 
+/obj/item/integrated_circuit/converter/findstring
+	name = "find text"
+	desc = "This gives position of sample in the string. Or returns 0."
+	extended_desc = "first pin is string,to be examined.Second pin is sample,which shoul be found. \
+	For example, 'eat this burger',' ' will give you position 4.This circuit isn't case sensitive."
+	complexity = 4
+	inputs = list(
+		"string" = IC_PINTYPE_STRING,
+		"sample" = IC_PINTYPE_STRING,
+		)
+	outputs = list(
+		"position" = IC_PINTYPE_NUMBER
+		)
+	activators = list("search" = IC_PINTYPE_PULSE_IN, "after search" = IC_PINTYPE_PULSE_OUT)
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+
+
+
+/obj/item/integrated_circuit/converter/findstring/do_work()
+
+	set_pin_data(IC_OUTPUT, 1, findtext(get_pin_data(IC_INPUT, 1),get_pin_data(IC_INPUT, 2)) )
+	push_data()
+
+	activate_pin(2)
+
+/obj/item/integrated_circuit/converter/exploders
+	name = "string exploder"
+	desc = "This splits single string by delimiter into list of strings."
+	extended_desc = "This circuits splits a given string into list of strings, based on the string, and given delimiter. \
```

Comment:
`This circuit splits a given string into a list of strings based on the string and given delimiter.`